### PR TITLE
Remove methods from Datasource/ConnectionInterface

### PR DESCRIPTION
**Pull request by ADmad**
These methods only make sense for RDBMS.

### CODE REVIEW

**Comment by ADmad** [comment]
If this seems reasonable than I would also like to update `ConnectionManager` and `ConnectionRegistry` to only accept/return `ConnectionInterface` instances, unless we want to keep open the possibility for more varied connection types in the future.

File: src/TestSuite/Fixture/FixtureDataManager.php [hunk 1]
```diff
@@ -205,7 +205,15 @@ protected function runOperation(array $fixtures, callable $operation): void
         $dbs = $this->fixtureConnections($fixtures);
         foreach ($dbs as $connection => $fixtures) {
             $db = ConnectionManager::get($connection);
+            if (
+                !method_exists($db, 'transactional')
+                || !method_exists($db, 'disableConstraints')
+            ) {
+                continue;
```
Comment (markstory) [inline src/TestSuite/Fixture/FixtureDataManager.php:212]: Should this be calling `$operation($db, $fixtures)` so that the operation still happens?

Comment (ADmad) [reply src/TestSuite/Fixture/FixtureDataManager.php]: Yup, fixed.

**Comment by ADmad** [comment]
Is this good to merge?

**Pull request merged**